src/Model/Command/UserRecommendation.php

```diff
@@ -0,0 +1,111 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\Matej\Model\Command;
+
+/**
+ * Deliver personalized recommendations for the given user.
+ */
+class UserRecommendation extends AbstractCommand
+{
+    const MINIMAL_RELEVANCE_LOW = 'low';
+    const MINIMAL_RELEVANCE_MEDIUM = 'medium';
+    const MINIMAL_RELEVANCE_HIGH = 'high';
+
+    /** @var string */
+    private $userId;
+    /** @var int */
+    private $count;
+    /** @var string */
+    private $scenario;
+    /** @var float */
+    private $rotationRate;
+    /** @var int */
+    private $rotationTime;
+    /** @var bool */
+    private $hardRotation;
+    /** @var string */
+    private $minimalRelevance = self::MINIMAL_RELEVANCE_LOW;
+    /** @var string */
+    private $filter = 'valid_to >= NOW';
```

Comment:
Isn`t this client specific.

`valid_to` must be present in item properties.

Reply:
Yes, its client specific.

Another argument is that 'filters' are client specific too (afaik all of them are still hardcoded).

So I suggests no default filter.

Reply:
Yes, it is client specific, but:
- all but one client ;-) will use it as for now, so we can make life easier for the vast majority  by defining the default filter like this;
- the one who does not use this default filter (but custome one), will anyway have to define the  filter -  so there will be no difference for him, even with this default filter, as it won't impact him.

From the beginning we planned to have a simple way how to overwrite the default filter for all recommendations and sorting - so there will be no extra work for those not using the default filter. But for those who do use the default we will make the library adoption simpler, IMO.

Also, quoting Matej official docs:
> Matej supports following filter: `valid_to >= NOW` which should be used in each recommendation request.

I will talk about this issue with RAD team and we may change this behavior in the future, however, having this feature in the API client version was partially their idea ¯\\\_(ツ)_/¯.

Reply:
There will be extra work for those who do not use default filter at all. The have to reset it. Or they end up with invalid request i guess.

It still seem weird to me to have default filter in such general library.

Reply:
No, they will not have to explicitly *reset* it. They will have to set custom filter anyway (and this will override the default one - if any). So no extra work for them. Because AFAIK there is no meaningful usecase for requesting recommendations without similar kind of default filter.

Reply:
I agree. It might be useful though to have a way how to set the *default* filter for the customers use-case.

How about having a static variable that'll be the default value?

```php
UserRecommendation::$DEFAULTS['filter'] = null;
```

How do other libs solve this?